packages/gdal/meta.yaml

```diff
@@ -0,0 +1,102 @@
+package:
+  name: gdal
+  version: 3.5.1
+
+source:
+  url: https://github.com/OSGeo/gdal/releases/download/v3.5.2/gdal-3.5.2.tar.gz
+  sha256: fbd696e1b2a858fbd2eb3718db16b14ed9ba82521d3578770d480c74fe1146d2
+requirements:
+  host:
+    - libtiff
+    - libproj
+    - libiconv
+    - geos
+  run:
+    - geos
+
+build:
+  sharedlibrary: true
+  script: |
+    export EMSCRIPTEN_SYSROOT=$(em-config CACHE)/sysroot
+    export EMSCRIPTEN_INCLUDE=$EMSCRIPTEN_SYSROOT/include
+    export EMSCRIPTEN_BIN=$EMSCRIPTEN_SYSROOT/bin
+    export EMSCRIPTEN_LIB=$EMSCRIPTEN_SYSROOT/lib/wasm32-emscripten/pic
+
+    mkdir -p dist
+    export DISTDIR=$(pwd)/dist
+
+    embuilder build zlib --pic
+    embuilder build libjpeg --pic
+    embuilder build libpng --pic
+    embuilder build sqlite3 --pic
```

Comment:
Do we need a second copy of sqlite3 for gdal? Or does this share the original one?

Reply:
It shares the original one. `embuilder build sqlite3` will explicitly build sqlite3 while adding `-sUSE_SQLITE3` flag will implicitly build it. But both shares the same library (built into emsdk/emsdk/emscripten/cache/sysroot/...).

I found that `-sUSE_...` flags are sometimes ignored (maybe it is related to CMake or https://github.com/emscripten-core/emscripten/issues/17675, not sure though). So I used `embuilder` to explicitly build those libraries.